.buildkite/generate_pipeline.py

```
@@ -0,0 +1,240 @@
+
+            clouds_to_include = []
+            clouds_to_exclude = []
+            is_serve_test = False
+            for decorator in node.decorator_list:
+                if isinstance(decorator, ast.Call):
+                    # We only need to consider the decorator with no arguments
+                    # to extract clouds.
+                    continue
+                full_path = _get_full_decorator_path(decorator)
+                if full_path.startswith('pytest.mark.'):
+                    assert isinstance(decorator, ast.Attribute)
+                    suffix = decorator.attr
+                    if suffix.startswith('no_'):
+                        clouds_to_exclude.append(suffix[3:])
+                    else:
+                        if suffix == 'serve':
+                            is_serve_test = True
+                            continue
+                        if suffix not in ALL_CLOUDS_IN_SMOKE_TESTS:
+                            # This mark does not specify a cloud, so we skip it.
+                            continue
+                        clouds_to_include.append(suffix)
+            clouds_to_include = (clouds_to_include if clouds_to_include else
+                                 copy.deepcopy(DEFAULT_CLOUDS_TO_RUN))
+            clouds_to_include = [
+                cloud for cloud in clouds_to_include
+                if cloud not in clouds_to_exclude
+            ]
+            cloud_queue_map = SERVE_CLOUD_QUEUE_MAP if is_serve_test else CLOUD_QUEUE_MAP
+            final_clouds_to_include = [
+                cloud for cloud in clouds_to_include if cloud in cloud_queue_map
+            ]
+            if clouds_to_include and not final_clouds_to_include:
+                print(f'Warning: {file_path}:{node.name} '
+                      f'is marked to run on {clouds_to_include}, '
+                      f'but we do not have credentials for those clouds. '
+                      f'Skipped.')
```

Do we expect this to happen in normal operation? I think in this case it's more preferable to throw an exception and crash, rather than silently ignore the cloud.

---

The test cases have:

```
@pytest.mark. runpod
def test_runpod():
    pass

@pytest.mark. aws
def test_aws():
    pass
```

In `generate_pipeline.py`, I want to keep `test_aws` and filter out `test_runpod` since we don't have the credentials to run `test_runpod`. That is the expected behavior.